reporter/src/main/kotlin/reporters/AbstractNoticeReporter.kt

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2017-2018 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.reporter.reporters
+
+import ch.frankel.slf4k.info
+
+import com.here.ort.model.OrtResult
+import com.here.ort.reporter.Reporter
+import com.here.ort.reporter.ResolutionProvider
+import com.here.ort.utils.log
+import com.here.ort.utils.spdx.getLicenseText
+
+import java.io.File
+import java.util.SortedMap
+import java.util.SortedSet
+
+abstract class AbstractNoticeReporter : Reporter() {
```

Comment:
AS far as I understand the difference between the future child classes of the abstract reported is just the filtering of the relevant licenses out of the given licenses. Why did you decide to solve this via inheritance instead of aggregation? 
e.g. did you consider having a single implementation which allows e.g. injecting a filter into the constructor?

Reply:
This is because we discover reporters and make them available on the command line by looking for implementations of the `Reporter` base class using `ServiceLoader`. Making one configurable class would not be compatible with this approach.

Reply:
Is this discovery mechanism actually used? If it is not is it really needed? why is it based on inheritance?

Reply:
We use the standard [ServiceLoader](https://docs.oracle.com/javase/8/docs/api/java/util/ServiceLoader.html) approach to discover any reporter classes on the classpath, to allow reporters being written as plugins.